tests/nest_tests/fir_filter_test.py

```diff
@@ -0,0 +1,153 @@
+# -*- coding: utf-8 -*-
+#
+# fir_filter_test.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import unittest
+
+import numpy as np
+
+try:
+    import matplotlib
+    import matplotlib.pyplot as plt
+
+    TEST_PLOTS = True
+except BaseException:
+    TEST_PLOTS = False
+
+import nest
+import scipy
+import scipy.signal
+import scipy.stats
+
+from pynestml.frontend.pynestml_frontend import to_nest, install_nest
+
+
+class NestFirFilterTest(unittest.TestCase):
+    """
+    Tests the working of FIR filter model in NEST
+    """
+
+    def test_fir_filter(self):
+        nestml_model_file = 'FIR_filter.nestml'
+        nestml_model_name = 'fir_filter_nestml'
+        target_path = '/tmp/fir-filter'
+        logging_level = 'INFO'
+        module_name = 'nestmlmodule'
+        store_log = False
+        suffix = '_nestml'
+        dev = True
+
+        # Generate the NEST code
+        input_path = os.path.join(os.path.realpath(os.path.join(os.path.dirname(__file__), 'resources', nestml_model_file)))
+        nest_path = nest.ll_api.sli_func("statusdict/prefix ::")
+        to_nest(input_path, target_path, logging_level, module_name, store_log, suffix, dev)
```

Comment:
I'm sure this has nothing to do with this PR, but this function call switches the matplotlib backend from `MacOSX` to `agg` on my system. Can't see a direct reason for that in `pynestml.frontend.pynestml_frontend`, but should I assume that this behaviour is intentional?

Reply:
The matplotlib backend was switched to `agg` by default in `ode-toolbox` which we use in NESTML for solving differential equations. It is now fixed in this [PR](https://github.com/nest/ode-toolbox/pull/50) and merged to master. Perhaps you could update your ode-toolbox with the latest master by issuing the following commands and test this again?
```
pip uninstall odetoolbox
pip install git+https://github.com/nest/ode-toolbox
```

Reply:
Thanks @pnbabu, updating odetoolbox fixes the matplotlib backend issue.